src/plugins/navigation/mutations/createNavigationTree.js

```diff
@@ -0,0 +1,60 @@
+import Random from "@reactioncommerce/random";
+import decodeNavigationTreeItemIds from "../util/decodeNavigationTreeItemIds.js";
+import setDefaultsForNavigationTreeItems from "../util/setDefaultsForNavigationTreeItems.js";
+import { NavigationTree as NavigationTreeSchema } from "../simpleSchemas.js";
+
+/**
+ * @method createNavigationTree
+ * @summary Creates a navigation tree
+ * @param {Object} context An object containing the per-request state
+ * @param {Object} input Input of createNavigationTree mutation
+ * @param {String} input.navigationTreeId ID of navigation tree to update
+ * @param {String} input.shopId Shop ID of navigation tree
+ * @param {Object} input.navigationTree Navigation tree object to update
+ * @returns {Promise<Object>} Created navigation tree
+ */
+export default async function createNavigationTree(context, input) {
+  const { collections } = context;
+  const { NavigationTrees } = collections;
+  const { shopId, navigationTree } = input;
+
+  const {
+    shouldNavigationTreeItemsBeAdminOnly,
+    shouldNavigationTreeItemsBePubliclyVisible,
+    shouldNavigationTreeItemsBeSecondaryNavOnly
+  } = await context.queries.appSettings(context, shopId);
+
+  // Navigation item visibility defaults from settings
+  const visibilityDefaults = {
+    shouldNavigationTreeItemsBeAdminOnly,
+    shouldNavigationTreeItemsBePubliclyVisible,
+    shouldNavigationTreeItemsBeSecondaryNavOnly
+  };
+
+  // Set default values for navigation tree draft items before validation.
+  // isVisible, isPrivate, and isSecondary are optional in the GraphQL schema,
+  // but are required by SimpleSchema. This reduces the input payload size by not
+  // needing to include values that aren't explicitly set.
+  const navigationTreeData = {
+    ...navigationTree,
+    shopId,
+    _id: Random.id(),
+    draftItems: setDefaultsForNavigationTreeItems(navigationTree.draftItems, visibilityDefaults)
+  };
+
+  // Validate the navigation tree with the defaults set
+  NavigationTreeSchema.validate(navigationTreeData);
+  const { draftItems } = navigationTreeData;
+
+  await context.validatePermissions("reaction:navigationTrees", "create", { shopId, legacyRoles: ["core"] });
+
+  if (draftItems) {
+    decodeNavigationTreeItemIds(draftItems);
```

Comment:
@trojanh I see you copied this from how it's done in the update, but it's incorrect. All ID decoding should happen in the GraphQL resolvers. This function should assume that all IDs are internal. Would you mind fixing it here and also for the `updateNavigationTree` mutation, by moving `decodeNavigationTreeItemIds` lines to the resolvers?

Reply:
sure @aldeed I was skeptical about this part but since it was used in `updateNavigationTree` I overlooked it. Thanks for correcting. 

I will fix this.